SpatialGDK/Source/SpatialGDK/Private/Interop/ClientNetLoadActorHelper.cpp

```diff
@@ -0,0 +1,132 @@
+		{
+			NetDriver->PackageMap->ResolveSubobject(DynamicSubObject, ObjectRef);
+			UE_LOG(LogClientNetLoadActorHelper, Verbose,
+				   TEXT("Found reusable dynamic SubObject (ObjectRef offset: %u) for ClientNetLoad actor with entityId %d"),
+				   ObjectRef.Offset, ObjectRef.Entity);
+			return DynamicSubObject;
+		}
+	}
+	return nullptr;
+}
+
+void FClientNetLoadActorHelper::EntityRemoved(const Worker_EntityId EntityId, const AActor& Actor)
+{
+	ClearDynamicSubObjectRefs(EntityId);
+	if (USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(EntityId))
+	{
+		for (UObject* DynamicSubObject : Channel->CreateSubObjects)
+		{
+			FNetworkGUID SubObjectNetGUID = NetDriver->PackageMap->GetNetGUIDFromObject(DynamicSubObject);
+			if (SubObjectNetGUID.IsValid())
+			{
+				FUnrealObjectRef SubObjectRef = NetDriver->PackageMap->GetUnrealObjectRefFromNetGUID(SubObjectNetGUID);
+				if (SubObjectRef.IsValid() && IsDynamicSubObject(*NetDriver, Actor, SubObjectRef.Offset))
+				{
+					SaveDynamicSubObjectRef(SubObjectRef, SubObjectNetGUID);
+					UE_LOG(
+						LogClientNetLoadActorHelper, Verbose,
+						TEXT("Saved reusable dynamic SubObject ObjectRef (ObjectRef offset: %u) for ClientNetLoad actor with entityId %d"),
+						SubObjectRef.Offset, SubObjectRef.Entity);
+				}
+			}
+		}
+	}
+}
+
+FNetworkGUID* FClientNetLoadActorHelper::GetSavedDynamicSubObjectNetGUID(const FUnrealObjectRef& ObjectRef)
+{
+	if (TMap<FUnrealObjectRef, FNetworkGUID>* EntityMappings = DynamicSubObjectRefToGuid.Find(ObjectRef.Entity))
+	{
+		if (FNetworkGUID* NetGUID = EntityMappings->Find(ObjectRef))
+		{
+			return NetGUID;
+		}
+	}
+	return nullptr;
+}
+
+void FClientNetLoadActorHelper::SaveDynamicSubObjectRef(const FUnrealObjectRef& ObjectRef, const FNetworkGUID& NetGUID)
+{
+	TMap<FUnrealObjectRef, FNetworkGUID>& EntityMappings = DynamicSubObjectRefToGuid.FindOrAdd(ObjectRef.Entity);
+	EntityMappings.Emplace(ObjectRef, NetGUID);
+}
+
+void FClientNetLoadActorHelper::ClearDynamicSubObjectRefs(const Worker_EntityId InEntityId)
+{
+	DynamicSubObjectRefToGuid.Remove(InEntityId);
+}
+
+void FClientNetLoadActorHelper::RemoveRuntimeRemovedComponents(const Worker_EntityId EntityId, const TArray<ComponentData>& NewComponents)
+{
+	auto ContainedInComponentsArr = [this, EntityId, &NewComponents](const FUnrealObjectRef& CheckComponentObjRef) {
+		for (const ComponentData& Component : NewComponents)
+		{
+			// Skip if this isn't a generated component
+			if (Component.GetComponentId() < SpatialConstants::STARTING_GENERATED_COMPONENT_ID)
+			{
+				continue;
+			}
+
+			uint32 Offset = 0;
+			NetDriver->ClassInfoManager->GetOffsetByComponentId(Component.GetComponentId(), Offset);
+
+			if (FUnrealObjectRef(EntityId, Offset) == CheckComponentObjRef)
+			{
+				return true;
+			}
+		}
+		return false;
+	};
+
+	if (TMap<FUnrealObjectRef, FNetworkGUID>* EntityMappings = DynamicSubObjectRefToGuid.Find(EntityId))
```

Comment:
Nit: Maybe we could use slightly more descriptive names?
Maybe even switch them? DynamicSubObjectRefToGuid on the left-hand side, and `EntityMappings` or `EntityDynamicSubobjectSOMETHING` on the right-hand side.

Reply:
yeee this is done now